v/V.g4

```diff
@@ -0,0 +1,1332 @@
+    }
+
+     /**
+     * Returns {@code true} if no line terminator exists between the specified
+     * token offset and the prior one on the {@code HIDDEN} channel.
+     *
+     * @return {@code true} if no line terminator exists between the specified
+     * token offset and the prior one on the {@code HIDDEN} channel.
+     */
+    private boolean noTerminatorBetween(int tokenOffset) {
+        BufferedTokenStream stream = (BufferedTokenStream)_input;
+        List<Token> tokens = stream.getHiddenTokensToLeft(stream.LT(tokenOffset).getTokenIndex());
+
+        if (tokens == null) {
+            return true;
+        }
+
+        for (Token token : tokens) {
+            if (token.getText().contains("\n"))
+                return false;
+        }
+
+        return true;
+    }
+
+     /**
+     * Returns {@code true} if no line terminator exists after any encounterd
+     * parameters beyond the specified token offset and the next on the
+     * {@code HIDDEN} channel.
+     *
+     * @return {@code true} if no line terminator exists after any encounterd
+     * parameters beyond the specified token offset and the next on the
+     * {@code HIDDEN} channel.
+     */
+    private boolean noTerminatorAfterParams(int tokenOffset) {
+        BufferedTokenStream stream = (BufferedTokenStream)_input;
+        int leftParams = 1;
+        int rightParams = 0;
+        String value;
+
+        if (stream.LT(tokenOffset).getText().equals("(")) {
+            // Scan past parameters
+            while (leftParams != rightParams) {
+                tokenOffset++;
+                value = stream.LT(tokenOffset).getText();
+
+                if (value.equals("(")) {
+                    leftParams++;
+                }
+                else if (value.equals(")")) {
+                    rightParams++;
+                }
+            }
+
+            tokenOffset++;
+            return noTerminatorBetween(tokenOffset);
+        }
+
+        return true;
+    }
+}
+
+@lexer::members {
+
+    // The most recently produced token.
+    private Token lastToken = null;
+
+    /**
+     * Return the next token from the character stream and records this last
+     * token in case it resides on the default channel. This recorded token
+     * is used to determine when the lexer could possibly match a regex
+     * literal.
+     *
+     * @return the next token from the character stream.
+     */
+    @Override
+    public Token nextToken() {
+
+        // Get the next token.
+        Token next = super.nextToken();
+
+        if (next.getChannel() == Token.DEFAULT_CHANNEL) {
+            // Keep track of the last token on the default channel.
+            this.lastToken = next;
+        }
+
+        return next;
+    }
+}
+
+//SourceFile       = ModuleClause ";" { ImportDecl ";" } { TopLevelDecl ";" } .
+sourceFile
+    : (moduleClause eos)? ( importDecl eos )* ( topLevelDecl eos )*
+    ;
+
+/*
+    V语言没有package关键字，取而代之的是 module 关键字
```

Comment:
I don't like using of not English language in comments. We do international grammars 😃  Here and below.